firefox-ios/Client/Frontend/Reader/ReaderMode.swift

```diff
@@ -109,7 +109,16 @@ enum ReaderModeFontSize: Int {
     }
 
     static var defaultSize: ReaderModeFontSize {
-        switch UIApplication.shared.preferredContentSizeCategory {
+        var contentSizeCategory: UIContentSizeCategory?
+        if Thread.isMainThread {
+            contentSizeCategory = UIApplication.shared.preferredContentSizeCategory
+        } else {
+            DispatchQueue.main.sync {
+                contentSizeCategory = UIApplication.shared.preferredContentSizeCategory
+            }
+        }
```

Comment:
instead of this, let's just use `ensureMainThread` to be in line with what we're using now, and call it a day! :)

Reply:
unfortunately 'ensureMainThread' is asynchronous.

Reply:
whoops, you're right. I saw glossed over the `.sync` part. lol. I'm a goof.

Reply:
You should never dispatch code onto the main thread synchronously, you can interrupt certain tasks that will crash the app if they become blocked in any way. 
Here is a decent article on the topic that covers why with some examples. https://sarawanak.medium.com/demystifying-dispatchqueue-main-sync-b59b77e4f502

Reply:
The problem is when we dispatch to main queue while we are already on main queue, then we produce a deadlock, but in our case we check if we are in main thread and set the contentSize, otherwise we dispatch synchronously to main. I know the code doesn't look good and I will change the implementation according to requested changes.

Reply:
It's still not a good idea to interrupt the main thread, you can interrupt a task that cannot become blocked and can result in crashes. Why does using async not work here?

Reply:
Given the synchronous nature of property getters and the need to return a value immediately, using async within them is not a good idea.

Reply:
In the last commit I moved the call to `ReaderModeStyle.defaultStyle()` to be async called on the main thread, just before webserver call.